components/AddDonation.js

```diff
@@ -0,0 +1,144 @@
+import styled from "styled-components";
+import { useState } from "react";
+import TextLengthCounter from "./Counter";
+
+export default function AddDonation({ createDonation }) {
+  const [isFormVisible, setIsFormVisible] = useState(false);
+  const [count, setCount] = useState(0);
+  const translationMap = {
+    fruitsVegetables: "Obst & Gemüse",
+    homemade: "Selbstgemacht",
+    pastry: "Backwaren",
+  };
+
+  function handleToggleForm() {
+    setIsFormVisible(!isFormVisible);
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const newDonationData = Object.fromEntries(formData);
+
+    createDonation(newDonationData);
+    setCount(0);
+    event.target.reset();
+    setIsFormVisible(!isFormVisible);
+  }
+
+  return (
+    <StyledDiv>
+      {!isFormVisible ? (
```

Comment:
A good guideline is to always ask for the positive variant, if possible. This may not make a big difference at first glance, but it makes it easier to follow. 

"if this is not true then... else..." vs. "if this is true then... else."

Reply:
thats exactly what coach felix suggested in the comment https://github.com/minatimar/capstone-project/pull/9#discussion_r1095627904